src/json.rs

```diff
@@ -1667,7 +1669,13 @@ impl<T: Iterator<Item = char>> Parser<T> {
                         self.bump();
                         return Ok(res);
                     },
-                    Some(c) => res.push(c),
```

Comment:
This may be best expressed as:

```rust
match self.ch {
    Some('"') => { ... }
    Some(c) if c.is_control() => return self.error(...),
    Some(c) => res.push(c),
    ...
}
```